azext_iot/operations/hub.py

```diff
@@ -2651,15 +2651,22 @@ def conn_str_getter(hub):
                 discovery, hub, policy_name, key_type, show_all, default_eventhub
             )
 
-        return [
-            {
-                "name": hub.name,
-                "connectionString": conn_str_getter(hub)
-                if show_all
-                else conn_str_getter(hub)[0],
-            }
-            for hub in hubs if hub.properties.state == IoTHubStateType.Active.value
-        ]
+        connectionStrings = []
+        for hub in hubs:
+            if hub.properties.state == IoTHubStateType.Active.value:
```

Comment:
We should indicate in help that this will only query active state Iot Hubs.

Reply:
It would be helpful to create a logger debug message if a hub is being skipped for not being active.